File: src/cs/Ssh/Algorithms/CompressionAlgorithm.cs
// Copyright (c) Microsoft. All rights reserved.

using System;

namespace Microsoft.DevTunnels.Ssh.Algorithms;

public abstract class CompressionAlgorithm : SshAlgorithm, IDisposable
{
	protected CompressionAlgorithm(string name)
		: base(name)
	{
	}

	public abstract Buffer Compress(Buffer input);

	public abstract Buffer Decompress(Buffer input);

	public void Dispose()
	{
		Dispose(true);
		GC.SuppressFinalize(this);
	}

	protected abstract void Dispose(bool disposing);
}

File: src/ts/ssh/algorithms/compressionAlgorithm.ts
//
//  Copyright (c) Microsoft Corporation. All rights reserved.
//

import { Buffer } from 'buffer';
import { SshAlgorithm } from './sshAlgorithm';

export interface CompressionAlgorithm extends SshAlgorithm {
	compress(data: Buffer): Buffer;
	decompress(data: Buffer): Buffer;
}
